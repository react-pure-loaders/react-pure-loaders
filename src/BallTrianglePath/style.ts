import { keyframes } from '@emotion/react';
import { balls, globalAnimation } from '../mixins';
import { PRIMARY_COLOR } from '../variables';
import { Grid } from '../types';

const amount = '50px';

const ballTrianglePath1 = keyframes({
  '100%': {
    transform: 'translate(0px, 0px)',
  },
  '33%': {
    transform: `translate(${parseInt(amount, 10) / 2}px, -${amount})`,
  },
  '66%': {
    transform: `translate(${amount}, 0px)`,
  },
});

const ballTrianglePath2 = keyframes({
  '100%': {
    transform: 'translate(0px, 0px)',
  },
  '33%': {
    transform: `translate(${parseInt(amount, 10) / 2}px, ${amount})`,
  },
  '66%': {
    transform: `translate(-${parseInt(amount, 10) / 2}px, ${amount})`,
  },
});

const ballTrianglePath3 = keyframes({
  '100%': {
    transform: 'translate(0px, 0px)',
  },
  '33%': {
    transform: `translate(-${amount}, 0px)`,
  },
  '66%': {
    transform: `translate(-${parseInt(amount, 10) / 2}px, -${amount})`,
  },
});

function ballTrianglePath() {
  const animations = [ballTrianglePath1, ballTrianglePath2, ballTrianglePath3];
  let i = 1;
  const grid: Grid = {};
  for (; i < 4; i += 1) {
    grid[`> div:nth-of-type(${i})`] = {
      animationDelay: '0',
      animationDuration: '2s',
      animationIterationCount: 'infinite',
      animationName: animations[i - 1],
      animationTimingFunction: 'ease-in-out',
    };
  }
  return grid;
}

export default (color?: string) => ({
  ...ballTrianglePath(),
  position: 'relative' as 'relative',
  transform: `translate(-${parseInt(amount, 10) / 1.667}px, -${parseInt(amount, 10) / 1.333}px)`,
  ' > div': {
    ...balls(color),
    ...globalAnimation(),
    border: `1px solid ${color || PRIMARY_COLOR}`,
    borderRadius: '100%',
    height: '10px',
    position: 'absolute' as 'absolute',
    width: '10px',

    '&:nth-of-type(1)': {
      top: amount,
    },

    '&:nth-of-type(2)': {
      left: `${parseInt(amount, 10) / 2}px`,
    },

    '&:nth-of-type(3)': {
      left: amount,
      top: amount,
    },
  },
});
