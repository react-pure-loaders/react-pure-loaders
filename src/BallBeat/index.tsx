/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallBeat({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
