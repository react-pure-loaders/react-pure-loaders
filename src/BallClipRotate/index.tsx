/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallClipRotate({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)} data-testid="ball-clip-rotate-container">
                <div/>
            </div>
    );
  }

  return null;
}
