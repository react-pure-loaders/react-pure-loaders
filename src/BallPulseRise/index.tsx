/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRise({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
