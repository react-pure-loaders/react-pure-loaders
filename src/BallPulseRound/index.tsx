/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRound({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
            </div>
    );
  }

  return null;
}
