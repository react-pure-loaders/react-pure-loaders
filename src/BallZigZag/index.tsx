/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZag({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
