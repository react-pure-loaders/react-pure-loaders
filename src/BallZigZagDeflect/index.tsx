/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZagDeflect({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
