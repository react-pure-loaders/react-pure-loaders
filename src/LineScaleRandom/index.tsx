/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
export default function LineScaleRandom({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div />
                <div />
                <div />
                <div />
            </div>
    );
  }

  return null;
}
