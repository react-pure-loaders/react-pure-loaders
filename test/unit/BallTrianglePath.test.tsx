import React from 'react';
import Chance from 'chance';
import { matchers } from '@emotion/jest';
import { render, cleanup } from '@testing-library/react';


import BallTrianglePath from '../../src/BallTrianglePath';
import { PRIMARY_COLOR } from '../../src/variables';


expect.extend(matchers);

const chance = new Chance();

describe('<BallTrianglePath>', () => {
    afterEach(cleanup);

    test('BallTrianglePath should match snapshot', () => {
        const { container } = render(<BallTrianglePath loading={true}/>);

        expect(container.firstChild).toMatchSnapshot();
    });

    test.skip('BallTrianglePath should have default color', () => {
        const { container } = render(<BallTrianglePath loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', PRIMARY_COLOR, { target: '> div' });
    });

    test.skip('BallTrianglePath should have given color', () => {
        const color = chance.color({ format: 'hex' });
        const { container } = render(<BallTrianglePath color={color} loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', color, { target: '> div' });
    });

    test('BallTrianglePath should have no children', () => {
        const { container } = render(<BallTrianglePath loading={false}/>);

        expect(container.firstChild).toBeNull();
    });

    test('BallTrianglePath should have three children', () => {
        const { container } = render(<BallTrianglePath loading={true}/>);

        expect(container.querySelectorAll('div')).toHaveLength(4);
    });
});
