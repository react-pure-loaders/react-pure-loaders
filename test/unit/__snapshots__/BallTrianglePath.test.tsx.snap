// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BallTrianglePath> BallTrianglePath should match snapshot 1`] = `
@keyframes animation-0 {
  100% {
    -webkit-transform: translate(0px, 0px);
    -moz-transform: translate(0px, 0px);
    -ms-transform: translate(0px, 0px);
    transform: translate(0px, 0px);
  }

  33% {
    -webkit-transform: translate(25px, -50px);
    -moz-transform: translate(25px, -50px);
    -ms-transform: translate(25px, -50px);
    transform: translate(25px, -50px);
  }

  66% {
    -webkit-transform: translate(50px, 0px);
    -moz-transform: translate(50px, 0px);
    -ms-transform: translate(50px, 0px);
    transform: translate(50px, 0px);
  }
}

@keyframes animation-1 {
  100% {
    -webkit-transform: translate(0px, 0px);
    -moz-transform: translate(0px, 0px);
    -ms-transform: translate(0px, 0px);
    transform: translate(0px, 0px);
  }

  33% {
    -webkit-transform: translate(25px, 50px);
    -moz-transform: translate(25px, 50px);
    -ms-transform: translate(25px, 50px);
    transform: translate(25px, 50px);
  }

  66% {
    -webkit-transform: translate(-25px, 50px);
    -moz-transform: translate(-25px, 50px);
    -ms-transform: translate(-25px, 50px);
    transform: translate(-25px, 50px);
  }
}

@keyframes animation-2 {
  100% {
    -webkit-transform: translate(0px, 0px);
    -moz-transform: translate(0px, 0px);
    -ms-transform: translate(0px, 0px);
    transform: translate(0px, 0px);
  }

  33% {
    -webkit-transform: translate(-50px, 0px);
    -moz-transform: translate(-50px, 0px);
    -ms-transform: translate(-50px, 0px);
    transform: translate(-50px, 0px);
  }

  66% {
    -webkit-transform: translate(-25px, -50px);
    -moz-transform: translate(-25px, -50px);
    -ms-transform: translate(-25px, -50px);
    transform: translate(-25px, -50px);
  }
}

.emotion-0 {
  position: relative;
  -webkit-transform: translate(-29.994001199760046px, -37.50937734433609px);
  -moz-transform: translate(-29.994001199760046px, -37.50937734433609px);
  -ms-transform: translate(-29.994001199760046px, -37.50937734433609px);
  transform: translate(-29.994001199760046px, -37.50937734433609px);
}

.emotion-0>div:nth-of-type(1) {
  -webkit-animation-delay: 0;
  animation-delay: 0;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-name: animation-0;
  animation-name: animation-0;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
}

.emotion-0>div:nth-of-type(2) {
  -webkit-animation-delay: 0;
  animation-delay: 0;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-name: animation-1;
  animation-name: animation-1;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
}

.emotion-0>div:nth-of-type(3) {
  -webkit-animation-delay: 0;
  animation-delay: 0;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-name: animation-2;
  animation-name: animation-2;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
}

.emotion-0>div {
  border-radius: 100%;
  height: 10px;
  margin: 2px;
  width: 10px;
  background-color: #ffffff;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  border: 1px solid #ffffff;
  position: absolute;
}

.emotion-0>div:nth-of-type(1) {
  top: 50px;
}

.emotion-0>div:nth-of-type(2) {
  left: 25px;
}

.emotion-0>div:nth-of-type(3) {
  left: 50px;
  top: 50px;
}

<div
  class="emotion-0"
>
  <div />
  <div />
  <div />
</div>
`;
